x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java

```diff
@@ -119,9 +128,27 @@ protected TransportPreviewTransformAction(
     }
 
     @Override
-    protected void doExecute(Task task, PreviewTransformAction.Request request, ActionListener<PreviewTransformAction.Response> listener) {
+    protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
         final ClusterState clusterState = clusterService.state();
         TransformNodes.warnIfNoTransformNodes(clusterState);
```

Comment:
I think it makes sense to make this a hard check, if the cluster lacks a transform node, preview should not be possible.

This sounds like a breaking change, but seems acceptable to me, as it makes no sense to call preview without a transform node. If there are concerns, we could make this check only on master.